File: tracks/java/src/Darts.java
/**
 * The Darts class is used to calculate the score in a game of darts based on
 * the coordinates of where a dart lands on the dartboard.
 */
public class Darts {

    /**
     * Default constructor for Darts.
     * This constructor doesn't perform any special operations and is provided
     * to create instances of this class if needed.
     */
    public Darts() {

    }

    /**
     * Calculates the score of a dart throw based on its x and y coordinates on the
     * dartboard.
     * The dartboard has different scoring zones:
     * - The bullseye (center) scores 10 points.
     * - The middle circle scores 5 points.
     * - The outer circle scores 1 point.
     * - Outside the outer circle scores 0 points.
     *
     * @param xOfDart the x-coordinate of the dart's landing position
     * @param yOfDart the y-coordinate of the dart's landing position
     * @return the score of the dart throw
     */
    public int score(double xOfDart, double yOfDart) {
        double d = Math.sqrt((xOfDart * xOfDart) + (yOfDart * yOfDart));
        int result = 0;
        if (d > 10) {
            result = 0;
        } else if (5 < d && d <= 10) {
            result = 1;
        } else if (1 < d && d <= 5) {
            result = 5;
        } else if (d <= 1) {
            result = 10;
        }
        return result;
    }
}

File: tracks/python/darts.py
"""
This module provides functionality for calculating a score based on the Euclidean distance
from the origin to a given point. It includes the `score` function, which takes the coordinates
of a point as input and returns a score based on its distance from the origin. The scoring
system is tiered, awarding different points for distances within specific ranges.

The module relies on the `math` library for calculating the Euclidean distance.
"""

import math


def score(x, y):
    """Calculate the score based on the distance from the origin (0, 0) to the point (x, y).

    This function computes the Euclidean distance from the origin to the point (x, y) and
    assigns a score based on this distance. The scoring is as follows:
    - A distance of 1 or less scores 10 points.
    - A distance more than 1 but up to 5 scores 5 points.
    - A distance more than 5 but up to 10 scores 1 point.
    - A distance more than 10 scores 0 points.

    :param x: float - The x-coordinate of the point.
    :param y: float - The y-coordinate of the point.
    :return: int - The score based on the distance of the point from the origin.
    """
    d = math.sqrt((x ** 2) + (y ** 2))
    if d <= 1:
        return 10
    if d <= 5:
        return 5
    if d <= 10:
        return 1
    return 0
